### ISSUE TITLE
find-and-execute needs echo selected command

### ISSUE THREAD

**Issue by qiuchengxuan** [fisherman/fzf#30]
I think function __fzf_find_and_execute could have some improvement like this
function __fzf_find_and_execute
  __fish_history | __fzfcmd +s -m $FZF_FIND_AND_EXECUTE_OPTS --tiebreak=index --toggle-sort=ctrl-r --height 40% | read -l select
  if not test -z $select
    fish_prompt; echo $select
    eval $select
    commandline -f repaint
  end
end

**Comment by jethrokuan** [comment]
Correct me if I'm wrong, but the nature of selecting from __fish_history ensures that `$select` is always non-empty, so the checks wouldn't do much. Also, `--height` is an invalid option.

```
fish_prompt; echo $select
```

Could you explain this line?

It's easier for me to understand what you're getting at if you let me know the limitations of the current implementation.

**Comment by qiuchengxuan** [comment]
I installed this plugin from fisherman which seems to be outdated, so just ignore the checks.

--height works, make this looks closer to reverse command search in bash/zsh.

the problem I want to say is, after command selected(by fish-find-and-execute), I'm unable to see the echo of the command I selected, nor can I find it in history through Ctrl+P, which made this not that convenient to use

But It seems that my modification doesn't solve the problem I mentioned, sadly.

**Comment by jethrokuan** [comment]
I don't see a `--height` option in the man page, and I tried adding the option in to no avail.

You're right, currently the command doesn't do the following:

1. Add the command executed to history
2. Show the command executed

And this should be fixed.

**Comment by qiuchengxuan** [comment]
fzf --version
0.16.4

man page:
       --height=HEIGHT[%]
              Display fzf window below the cursor with the given height instead of using the full screen.
perhaps you need upgrade your fzf?

**Comment by jethrokuan** [comment]
I was indeed using an older version of fzf, updating it allowed for the height option. This can be added using the customizable variables listed in the README.

I took a look at adding the executed command to fish_history, but fish doesn't support this out of the box. One hacky way to do it is detailed here:

http://stackoverflow.com/questions/30813606/how-to-add-entry-to-fish-shell-history

However, I'm against adding this to the repo, because it would make things a lot less amenable to maintenance. I added an echo for the executed function though, for clarity.
